src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -191,21 +209,50 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
             handlerName: handlerName,
             originalHandlerName: handlerName,
             documentUri: documentUri,
-            samTemplatePath: templateInvoke?.samTemplatePath,
-            originalSamTemplatePath: templateInvoke?.samTemplatePath,
+            samTemplatePath: pathutil.normalize(templateInvoke?.samTemplatePath),
+            originalSamTemplatePath: pathutil.normalize(templateInvoke?.samTemplatePath),
             debugPort: config.noDebug ? -1 : await getStartPort(),
         }
 
+        //
+        // Configure and launch.
+        //
+        // 1. prepare a bunch of arguments
+        // 2. do `sam build`
+        // 3. do `sam local invoke`
+        //
         switch (runtimeFamily) {
             case RuntimeFamily.NodeJS:
-                launchConfig = await this.launchTypescript(launchConfig)
+                {
+                    const c: NodejsDebugConfiguration = await tsDebug.makeTypescriptConfig(launchConfig)
+                    launchConfig = c
+                    if (!noInvoke) {
+                        await tsDebug.invokeTypescriptLambda(this.ctx, c)
+                    }
+                }
                 break
-            case RuntimeFamily.Python:
-                launchConfig = await this.launchPython(launchConfig)
+            case RuntimeFamily.Python: {
+                //  Make a Python launch-config from the generic config.
+                const c: PythonDebugConfiguration = await pythonDebug.makePythonDebugConfig(
+                    launchConfig,
+                    !launchConfig.noDebug,
```

Comment:
A few things here:
* Seems weird to negate this. Can you refactor to use `launchConfig.isDebug` instead?
* Also, the `aws-sam` launch config will run EVERYTHING as a debug.
* Why do we need to specify all of these values if we're already passing `launchConfig` to the function? Can we just pull everything at that level?

Reply:
> Seems weird to negate this. Can you refactor to use `launchConfig.isDebug` instead?

goal is to eliminate `isDebug` field. `noDebug` field is a vscode-provided field, so we should use that instead of adding similar, redundant fields.

> Also, the `aws-sam` launch config will run EVERYTHING as a debug.

? We don't always attach a debugger.


>      Why do we need to specify all of these values if we're already passing `launchConfig` to the function? Can we just pull everything at that level?

The work here is iterative. There is a lot of low-hanging fruit.

Reply:
> 
> > Seems weird to negate this. Can you refactor to use `launchConfig.isDebug` instead?
> 
> goal is to eliminate `isDebug` field. `noDebug` field is a vscode-provided field, so we should use that instead of adding similar, redundant fields.
> 
> > Also, the `aws-sam` launch config will run EVERYTHING as a debug.
> 
> ? We don't always attach a debugger.

Forgot that VS Code has an inherent run-without-debug mode: https://code.visualstudio.com/docs/editor/debugging#_run-mode . I was saying that [our spec](https://github.com/aws/aws-toolkit-vscode/blob/master/designs/sam-debugging/local-sam-debugging.md) has nowhere to define a run-without-debug, so the only way to do it going forward will be with Ctrl+F5.

Reply:
> I was saying that [our spec](https://github.com/aws/aws-toolkit-vscode/blob/master/designs/sam-debugging/local-sam-debugging.md) has nowhere to define a run-without-debug, so the only way to do it going forward will be with Ctrl+F5.

User invokes the `Run: Start without debugging` command. That is the standard way to run a `launch.json` without debugging.

Our spec doesn't need to mention that because it's part of the editor conventions.